tests/codemods/test_file_resource_leak.py

```diff
@@ -0,0 +1,166 @@
+from tests.codemods.base_codemod_test import BaseCodemodTest
+from core_codemods.file_resource_leak import FileResourceLeak
+from textwrap import dedent
+
+
+class TestFileResourceLeak(BaseCodemodTest):
+    codemod = FileResourceLeak
+
+    def test_name(self):
+        assert self.codemod.name() == "file-resource-leak"
+
+    def test_simple(self, tmpdir):
+        input_code = """\
+        file = open('test.txt', 'r')
+        file.read()
+        """
+        expected = """\
+        with open('test.txt', 'r') as file:
+            file.read()
+        """
+        self.run_and_assert(tmpdir, dedent(input_code), dedent(expected))
+        assert len(self.file_context.codemod_changes) == 1
+
+    def test_simple_annotated(self, tmpdir):
+        input_code = """\
+        file: int = open('test.txt', 'r')
+        file.read()
+        """
+        expected = """\
+        with open('test.txt', 'r') as file:
+            file.read()
+        """
+        self.run_and_assert(tmpdir, dedent(input_code), dedent(expected))
+        assert len(self.file_context.codemod_changes) == 1
+
+    def test_just_open(self, tmpdir):
+        # strange as this change may be, it still leaks if left untouched
+        input_code = """\
+        file = open('test.txt', 'r')
+        """
+        expected = """\
+        with open('test.txt', 'r') as file:
+            pass
+        """
+        self.run_and_assert(tmpdir, dedent(input_code), dedent(expected))
+        assert len(self.file_context.codemod_changes) == 1
+
+    def test_multiple_assignments(self, tmpdir):
+        input_code = """\
+        file = file2 = open('test.txt', 'r')
+        file.read()
+        """
+        expected = """\
+        with open('test.txt', 'r') as file, file as file2:
+            file.read()
+        """
+        self.run_and_assert(tmpdir, dedent(input_code), dedent(expected))
+        assert len(self.file_context.codemod_changes) == 1
+
+    def test_minimal_block(self, tmpdir):
+        input_code = """\
+        file = open('test.txt', 'r')
+        file.read()
+        pass
+        """
+        expected = """\
+        with open('test.txt', 'r') as file:
+            file.read()
+        pass
+        """
```

Comment:
From this test I gather that the logic here is to indent everything after open call up to the read call, is that right?
So
```
var = 1
file = open('test.txt', 'r')
var = 2
file.read()
pass
```

would result in `var = 2` also being in the indented block, even if author didn't want that.

Reply:
We wrap all the statements until the last statement that references the `with` variable.
The situation in your example is deliberate. We don't really want to change the order of variables willy nilly.
Scope wise, having it inside the `with` block is fine as python has function level scope, that is, after the transformation, `var` will still be reachable anywhere else.